### PR TITLE
Surface extras and group conflicts in `uv export`

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Closes https://github.com/astral-sh/uv/issues/9364.

### CODE REVIEW

**Comment by BurntSushi** [comment]
I'm actually not sure `uv export` can support conflicting groups at all. Or at least, not all cases and not to `requirements.txt`. Because with my ongoing work to fix #9289, some cases can generate ambiguous dependencies. And since PEP 508 markers aren't expressive enough to capture the conditional logic required to disambiguate them, exporting to a flat format like `requirements.txt` that only supports PEP 508 markers could result in a set of requirements where there are multiple versions of the same package.

**Comment by charliermarsh** [comment]
Isn't this right, then? We only export a specific set of enabled groups and extras, and this PR ensures that they're all compatible.

**Comment by charliermarsh** [comment]
(`uv export` doesn't attempt to export all groups and extras; it exports a specific set of groups and extras.)

File: crates/uv/tests/it/lock.rs [hunk 1]
```diff
@@ -2360,6 +2360,15 @@ fn lock_conflicting_extra_basic() -> Result<()> {
     exit_code: 2
     ----- stdout -----
 
+    ----- stderr -----
+    error: extra `project1`, extra `project2` are incompatible with the declared conflicts: {`project[project1]`, `project[project2]`}
```
Comment (zanieb) [inline crates/uv/tests/it/lock.rs:2364]: I know this isn't from this pull request, but can we call these extras something other than `project` e.g., `extra1` and `extra2` for clarity? I changed these in the documentation but didn't realize there were tests with the same cases.

**Review by BurntSushi — APPROVED** [review]
> Isn't this right, then? We only export a specific set of enabled groups and extras, and this PR ensures that they're all compatible.

Hmmm yeah. I think you're right. False alarm on my part. Sorry!

File: crates/uv/tests/it/lock.rs (hunk 1 above)
Comment (charliermarsh) [reply crates/uv/tests/it/lock.rs:2364]: Yeah sure.

Comment (charliermarsh) [reply crates/uv/tests/it/lock.rs:2364]: Per other PR, I'll do this in a dedicated change, since a lot of other tests do this too.

**Pull request merged**